### DISCUSSION TITLE
file reader filter

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6237] [Q&A] [answered]
Is there any way to check for an empty file and not process when using the source as a file reader? I'm assuming I might need to do this in the preprocessor since any attempt to do do anything like checking msg.length throws an error due to hl7 to xml conversion.

**Comment by pacmano1** [comment, marked as answer]
At the bottom of the file reader you can specify min and max file size.  Does that work for you?

**Comment by rutledj** [comment]
That does work. Just wish the after processing action would still fire to delete the empty file.

**Reply by pacmano1** [reply]
I set up a scheduled job to delete (or move) files older than some X days.  You can do that in Mirth of course or at the operating system level.
